forgerock-authenticator/src/main/java/org/forgerock/android/auth/FRAClient.java

```diff
@@ -211,6 +211,17 @@ public boolean removeMechanism(@NonNull Mechanism mechanism) {
         return this.authenticatorManager.getAllNotifications(mechanism);
     }
 
+    /**
+     * Get the PushNotification object with its id. Identifier of PushNotification object is "<mechanismUUID>-<timeAdded>"
+     * Returns {@code null} if PushNotification could not be found.
```

Comment:
I think this will throw NPE instead of null if not found.

Reply:
There was a NullPointException caused by the attempt on set the PushMechanism on the PushNotification object. It was changed to check the null.